conans/client/rest/rest_client.py

```diff
@@ -340,21 +363,6 @@ def _remove_package_files(self, package_reference, files):
                                                          package_reference.package_id)
         return self._post_json(url, payload)
 
-    def server_info(self):
```

Comment:
The version checks were disabled (only log) since we use Artifactory and Bintray, so I removed them. The server is still sending it, but I ignore the headers in the client.